Marlin/Max7219_Debug_LEDs.h

```diff
@@ -69,18 +76,72 @@ void Max7219_LED_On(const uint8_t x, const uint8_t y);
 void Max7219_LED_Off(const uint8_t x, const uint8_t y);
 void Max7219_LED_Toggle(const uint8_t x, const uint8_t y);
 
-// Set all 8 LEDs in a single column
-void Max7219_Set_Column(const uint8_t col, const uint8_t val);
+// Set all LEDs in a single column
+void Max7219_Set_Column(const uint8_t col, const uint32_t val);
 void Max7219_Clear_Column(const uint8_t col);
 
-// Set all 8 LEDs in a single row
-void Max7219_Set_Row(const uint8_t row, const uint8_t val);
+// Set all LEDs in a single row
+void Max7219_Set_Row(const uint8_t row, const uint32_t val);
 void Max7219_Clear_Row(const uint8_t row);
 
+// 16 and 32 bit versions of Row and Column functions
+// Multiple rows and columns will be used to display the value if
+// the array of matrix LED's is too narrow to accomplish the goal
+void Max7219_Set_Rows_16bits(const uint8_t y, uint32_t val);
+void Max7219_Set_Rows_32bits(const uint8_t y, uint32_t val);
+void Max7219_Set_Columns_16bits(const uint8_t x, uint32_t val);
+void Max7219_Set_Columns_32bits(const uint8_t x, uint32_t val);
+
 // Quickly clear the whole matrix
 void Max7219_Clear();
 
 // Apply custom code to update the matrix
 void Max7219_idle_tasks();
 
+#ifndef MAX7219_ROTATE
+  #define MAX7219_ROTATE 0
+#endif
+#define _ROT ((MAX7219_ROTATE + 360) % 360)
+#if _ROT == 0
+  #define MAX7219_UPDATE_AXIS     y   // Fast line update axis for this orientation of the matrix display
+  #define MAX7219_X_LEDS          (8 * MAX7219_NUMBER_UNITS)
+  #define MAX7219_Y_LEDS          8
+  #define XOR_7219(x, y)          LEDs[y + (x >> 3) * 8] ^= _BV(7 - (x & 0x07))
+  #define SET_PIXEL_7219(x, y)    LEDs[y + (x >> 3) * 8] |= _BV(7 - (x & 0x07)) 
```

Comment:
To get slightly fewer shifts, `(x >>3) * 8` is the same as `((x & 0x100) << 1)` ?

Reply:
Yeah...   I'm starting to trust the compiler more and more.   Usually on this stuff...  GCC is smart enough and does something pretty reasonable.